packages/assets-controllers/src/NftController.ts

```diff
@@ -1526,10 +1479,6 @@ export class NftController extends BaseController<
 
     const checksumHexAddress = toChecksumHexAddress(tokenAddress);
 
-    // TODO: revisit this with Solana support and instead of passing chainId, make sure chainId is read from nftMetadata
-    const chainIdToAddTo =
-      chainId || this.#getCorrectChainId({ networkClientId });
-
```

Comment:
Removed this in favor of the mandatory `networkClientId`;
This will be called from `nftDetectionController` to add NFTs in which case we will take the chainId from the NFT and find the networkClientId. Or directly from clients, in which case they need to provide the networkClientId.